### ISSUE TITLE
[REQ] Elementor pro feature: Global Design Module Template

### ISSUE THREAD

**Issue by OCpublishing** [pojome/elementor#5252]
- [x] I have searched for similar features requests in both open and closed tickets and cannot find a duplicate.
- [x] The feature is still missing in the latest stable version of Elementor ( Elementor Pro. )

PLEASE, I spend some times editing this so be kind enough to read it full and of course feel free to ask anything if I am not clear enough

I asked it before but maybe I was not crystal clear and the answer pointed me to Elementor Pro theme builder form v2.0 single post template in a single line. But this is absolutly not what I am requesting 

**Request:**
A Global Design Module Template

**What it would do:**

- Creation and usage would be identical to templates inserted from library (not from template widget as we cannot edit the content using this method)
- Content could be edited separatly for each insertion in the same document (4 times in a single page for example), as the current template works now if inserted from folder/library.
- **BUT The design would be global**. So if you change the background color in the _'global design template'_ it will update everywhere it has been used before.

**Limitations right now:**
global widget: 
- Only  for widgets and it updates design AND content

Templates: 
- If inserted from library, specific text can be inserted each time you insert it in a page / post (you can insert it 2, 3, x times in a page /post and still be able to edit content for each one) but design cannot be updated globaly
- if Design can be updated but content cannot be modified.

**Example how it would be used:**
For example, lets say that I want to create a page with Chicken recipes.
I will create my page using elementor and every two or three paragraphs I want to add A _Global Design Module Template_ with a specific design that I could fill with my own content.

Title
Text
Module 1: fried chicken - Global design/ specific content
text 
Module 2 : Chicken à la sauce bearnaise - Global design/ specific content
texte
Module 3 : Chicken salad - Global design/ specific content
..........

The point of this is that I dont want to edit and modify manually the design of ALL the recipe modules
I created in my whole website one by one.
I just go to my 
_'global design template'_ and modify the design there and it is updated in my whole website

As you can see a single post template does not do the job because it has a definitive place and number of specific _modules_

Here is an example of the sections I designed.
![capture](https://user-images.githubusercontent.com/41859489/43366329-9d206708-933c-11e8-89b7-2ca2626a08af.JPG)

I want to be able to put it where I want, the number the time I want, with a specific content everytime, but the design would be saved under the _'global design template'_ and when i change the original template design it updates everywhere.

Think of it as a 'CSS design template for placeholders'

I hope I was more clear this time

I dont know if it would be a huge work for you guys, but I bet this could please a LOT of people. I searched for something that could do that and beside lots of copy paste CSS and HTML, there is no easy solution.

**Comment by Hanleymade** [comment]
Great idea!

**Comment by willcitizen** [comment]
Just want to throw in my 2c here as I have been putting a lot of thought into ways the Themebuilder could be improved and the gaps I find I want closed for my own desired way of working.  I think that your proposal is on the right track, but perhaps a little over-engineered in the implimentation.  What I kept coming back to here was that Elementor already has the concept of colour schemes and font schemes it's just not fully implemented (yet).
 If you've ever looked at the Gantry framework under the hood they use a concept of 'themes' which are really just sass color/font scheme templates that can be applied site-wide.  
All that elementor really lacks to be able to achieve (and I suspect the implimentation could almost be trivial) is firstly they need to flesh out the list of selectable colours and fonts in the global schemes to make them more usable.  
Once this is done all that needs to happen is the controls for style selection need to be made dynamic and the variables available in elementors global schemes exposed as part of the select list (this would also mean that you could hack out any combination of your own global style management solution - like creating your own 'themes' post type with ACF and building out swappable styles, or whatever works for you).  
Another option in terms of an elegant implementation of this would be to expand the copy/paste style functionality with a 'save global style' function and then add a style picker onto every element - you can then have a reusable library of global styles, centrally editable and you use the same edit/unlink workflow currently used on global widgets to manage, apply, or modify styles at a more specific level if needed.

**Comment by evanre** [comment]
+1. Unfortunately there is no such feature, even in PRO version. (Global widgets is different, I need a global section template!).
ps. I found a plugin which solves my problem - Anywhere Elementor. Waiting for core implementation

**Comment by erfan-ilyas** [comment]
Hello Everyone,

It's not relevant to idea that @OCpublishing suggested.

But as @willcitizen thought we might have something very close to using Global Styles with SCSS.

Please checkout:

https://wordpress.org/plugins/rabbitbuilder-global-central-js-css/

Hope it helps.

**Comment by vovkasolovev** [comment]
I support the proposed suggestions and want to offer a small addition to the ideas expressed about global templates here to keep all in one place. 

It would be nice to have a global fine-tuning of fonts and indentation between lines to customize the display of text in WYSIWYG widgets and in all kinds of special widgets. I see it as tuning a set of all kinds of combinations like
```
<h1>Lorem</h1>
<h2>Lorem</h2>
<p>Lorem</p>
<h2>Lorem</h2>
<h3>Lorem</h3>
<p>Lorem</p>
…
```
Font define design in a global sense. Visual editor for global font settings would solve most of the design work tasks that are now solved by using third-party themes or working with CSS the old-fashioned way. Because of the appearance of Theme Builder and Hello Theme it may be a very logical step in development.

**Comment by kneops** [comment]
+1

I definitely need this too, and on my first website ever to build with Elementor...
And to me, this should be part of any template builder.

This kind of global design module template should be able to be part of a any template, but marked as editable. 

The way I see it, it should work like this:

I create a page or single page/post template with a widget (doesn't matter what widget). Some of the widgets should be locked, some I could mark as editable. But only the content, not the design. So when a place and design a tabs widget and I mark it as editable within a template, the tabs title, icon and content can be changed and tabs can be added or deleted, but not it's design (spacing,colors, fonts etc.).

When a user/customer opens such a page in the Elementer editor, he/she can click 'Edit tabs', but not other parts of the tempate that are marked as Locked.

**Comment by elasticeye** [comment]
+1 for this

**Comment by srcek** [comment]
It would be great if such a template could be dynamic and used as a skin for Posts and Archive Posts widgets.

**Comment by LauGau** [comment]
Agree with @srcek 
I am currently doing all the layout I need with [Dynamic.ooo](https://www.dynamic.ooo/) even for products like I've done here at the bottom of the page : [Surfcamp listing](https://sonicesurfschool.com/surfcamp-ados/)

It will definitly be great to have something similiar natively in Elementor.

**Comment by jazir5** [comment]
This sounds like the features of Anywhere Elementor, and they would be great to have natively.

**Comment by pswerbung** [comment]
**+1 for this, very important** 
… to have a «global styles» using the template functionality and (of course … ) editing content later without loosing the (global) style. Basic of CSS of course, but in 2019 should be elementary for elementor too, right?

**Comment by black-eye** [comment]
+1

**Comment by princesshorsepower** [comment]
+1
Exactly, the Global Templates are pretty useless at the moment. There should be an option to only save the Style in the template, but let the content still be editable.

**Comment by MichaelDarko** [comment]
+1

**Comment by seDomhag** [comment]
Thank you  @OCpublishing for this really well written summary of the problem. We have exactly the same issue and are desperately searching for a solution to distinguish style settings from content settings on a global level.  We really hope that there will be a soution from Elementor (please guys!)

**Comment by arielk** [comment]
ref: https://github.com/elementor/elementor/issues/965

**Comment by tekonemedia** [comment]
+1 We need this terribly bad
